Plumb in the balance in the console wallet

philipr-za opened the pull request:
## Description
This PR plumbs in the actual wallet balance to the balance UI component in `tari_console_wallet`. The balance was added to the app_state and a refresh is triggered whenever there is any update to a transaction in the app_state.

## How Has This Been Tested?
Manual Testing

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I ran `cargo test` successfully before submitting my PR.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

StriderDM reviewed (dismissed):
LGTM, the clippy errors should be fixed in latest dev.

philipr-za committed:
Plumb in the balance in the console wallet

This PR plumbs in the actual wallet balance to the balance UI component in `tari_console_wallet`. The balance was added to the app_state and a refresh is triggered whenever there is any update to a transaction in the app_state.

philipr-za commented:
Rebased

delta1 reviewed (approved):
LGTM tested and working
